src/test/java/com/alipay/oceanbase/rpc/ObTableClientTest.java

```diff
@@ -2689,4 +2691,571 @@ public void testAggregationWithIllegalColumn() throws Exception {
         }
     }
+
+    @Test
+    // Test auto increment on rowkey
+    public void testAutoIncrementRowkey() throws Exception {
+        if (ObGlobal.OB_VERSION <= 0) {
+            // ob version is invalid
+            Assert.assertTrue(false);
+        } else if (ObGlobal.OB_VERSION == 4) {
+            // todo: auto increment only support in 4.x currently
+            final ObTableClient client = (ObTableClient) this.client;
+            client.addRowKeyElement("test_auto_increment_rowkey", new String[] { "c1", "c2" });
+
+            // use sql to create table
+            Connection connection = ObTableClientTestUtil.getConnection();
+            Statement statement = connection.createStatement();
+
+            try {
+                statement.execute("CREATE TABLE IF NOT EXISTS `test_auto_increment_rowkey`(c1 int auto_increment,c2 int not null,c3 int default null,primary key(c1, c2));");
+
+                client.insert("test_auto_increment_rowkey", new Object[] { 0, 1 }, new String[] { "c3"},
+                        new Object[] { 1 });
+
+                TableQuery tableQuery = client.query("test_auto_increment_rowkey");
+                tableQuery.select("c1", "c2", "c3");
+                ObTableValueFilter filter = new ObTableValueFilter(ObCompareOp.EQ, "c1", 1);
+                tableQuery.setFilter(filter);
+                QueryResultSet result = tableQuery.execute();
+                result.next();
+                Map<String, Object> value = result.getRow();
+                Assert.assertEquals(1, value.get("c2"));
+                Assert.assertEquals(1, value.get("c3"));
+
+                // test insert use user value
+                client.insert("test_auto_increment_rowkey", new Object[] { 100, 1 }, new String[] { "c3"},
+                        new Object[] { 1 });
+
+                tableQuery.select("c1", "c2", "c3");
+                filter = new ObTableValueFilter(ObCompareOp.EQ, "c1", 100);
+                tableQuery.setFilter(filter);
+                result = tableQuery.execute();
+                result.next();
+                value = result.getRow();
+                Assert.assertEquals(1, value.get("c2"));
+                Assert.assertEquals(1, value.get("c3"));
+
+                // test insert sync global auto inc val
+                client.insert("test_auto_increment_rowkey", new Object[] { 0, 1 }, new String[] { "c3"},
+                        new Object[] { 1 });
+
+                tableQuery.select("c1", "c2", "c3");
+                filter = new ObTableValueFilter(ObCompareOp.EQ, "c1", 101);
+                tableQuery.setFilter(filter);
+                result = tableQuery.execute();
+                result.next();
+                value = result.getRow();
+                Assert.assertEquals(1, value.get("c2"));
+                Assert.assertEquals(1, value.get("c3"));
+
+                // test delete
+                client.delete("test_auto_increment_rowkey", new Object[] { 101, 1 });
```

Comment:
I guess we should check whether delete success or not in this test.

Reply:
fixed